jbpm-flow/src/main/java/org/jbpm/workflow/core/impl/WorkflowProcessImpl.java

```diff
@@ -116,4 +116,30 @@ protected void processNodeContainer(org.jbpm.workflow.core.NodeContainer nodeCon
             }
         }
     }
+
+    protected void getParentNodes(Node currentNode, org.jbpm.workflow.core.NodeContainer nodeContainer, List<Node> nodes, long nodeId) {
+        for (Node node : nodeContainer.getNodes()) {
+            if (nodeId == node.getId()) {
+                if (currentNode != null) {
+                    nodes.add(currentNode);
+                }
+                break;
+            } else {
+                if (node instanceof org.jbpm.workflow.core.NodeContainer) {
+                    getParentNodes(node, (org.jbpm.workflow.core.NodeContainer) node, nodes, nodeId);
+                }
+            }
+        }
+    }
+
+    public Node getParentNode(long nodeId) {
+        List<Node> nodes = new ArrayList<>();
+
+        getParentNodes(null, nodeContainer, nodes, nodeId);
+
+        if (nodes.size() > 0) {
+            return nodes.get(0);
```

Comment:
how do we know it's the right node? :)

Reply:
Just picking the first ? :) Anyway, now the function return the first node that contain a node with the passed nodeId. I hope that would be the good one..